sig-and-wg/sig-core/proposals/ui-api-layer-modularization.md

```diff
@@ -0,0 +1,88 @@
+# UI API Layer Modularization
+
+Created on 2018-29-11 by Paweł Kosiec (@pkosiec).
+
+## Status
+
+Proposed on 2018-11-29.
+
+## Motivation
+
+UI API Layer exposes GraphQL API for the Kyma Console. It consists of resolver logic for different domains, like Kubernetes, Service Catalog, Remote Environments, Kubeless, etc. As Kyma needs to be modularized, users should be able to install only the components they need.
+
+Current approach we use in UI API Layer makes it impossible. During the GraphQL server start, we try to synchronize Stores of all Informers. In a result, if a resource of a specific type doesn't exist, UI API Layer won't start properly, because its Informer will return an error.
```

Comment:
```suggestion
The current approach we use in UI API Layer makes it impossible. During the GraphQL server start, we try to synchronize Stores of all Informers. As a result, if a resource of a specific type doesn't exist, UI API Layer won't start properly because its Informer will return an error.
```